pkg/controller/chaosengine/chaosengine_controller.go

```diff
@@ -512,33 +494,49 @@ func (r *ReconcileChaosEngine) getChaosEngineInstance(request reconcile.Request)
 }
 
 // Get application details
-func getApplicationDetail(ce *chaosTypes.EngineInfo) (*chaosTypes.EngineInfo, error) {
+func getApplicationDetail(engine *chaosTypes.EngineInfo) error {
 	applicationInfo := &chaosTypes.ApplicationInfo{}
-	appInfo, err := initializeApplicationInfo(ce.Instance, applicationInfo)
+	appInfo, err := initializeApplicationInfo(engine.Instance, applicationInfo)
 	if err != nil {
-		return ce, err
+		return err
 	}
-	ce.AppInfo = appInfo
+	engine.AppInfo = appInfo
 
 	var appExperiments []string
 	for _, exp := range appInfo.ExperimentList {
 		appExperiments = append(appExperiments, exp.Name)
 	}
-	ce.AppExperiments = appExperiments
+	engine.AppExperiments = appExperiments
 
 	chaosTypes.Log.Info("App key derived from chaosengine is ", "appLabelKey", chaosTypes.AppLabelKey)
 	chaosTypes.Log.Info("App Label derived from Chaosengine is ", "appLabelValue", chaosTypes.AppLabelValue)
 	chaosTypes.Log.Info("App NS derived from Chaosengine is ", "appNamespace", appInfo.Namespace)
 	chaosTypes.Log.Info("Exp list derived from chaosengine is ", "appExpirements", appExperiments)
-	chaosTypes.Log.Info("Monitoring Status derived from chaosengine is", "monitoringStatus", ce.Instance.Spec.Monitoring)
-	chaosTypes.Log.Info("Runner image derived from chaosengine is", "runnerImage", ce.Instance.Spec.Components.Runner.Image)
-	chaosTypes.Log.Info("exporter image derived from chaosengine is", "exporterImage", ce.Instance.Spec.Components.Monitor.Image)
-	chaosTypes.Log.Info("Annotation check is ", "annotationCheck", ce.Instance.Spec.AnnotationCheck)
-	return ce, nil
+	chaosTypes.Log.Info("Monitoring Status derived from chaosengine is", "monitoringStatus", engine.Instance.Spec.Monitoring)
+	chaosTypes.Log.Info("Runner image derived from chaosengine is", "runnerImage", engine.Instance.Spec.Components.Runner.Image)
+	chaosTypes.Log.Info("exporter image derived from chaosengine is", "exporterImage", engine.Instance.Spec.Components.Monitor.Image)
+	chaosTypes.Log.Info("Annotation check is ", "annotationCheck", engine.Instance.Spec.AnnotationCheck)
+	return nil
 }
 
 // Check if the engineRunner pod already exists, else create
-func (r *ReconcileChaosEngine) checkEngineRunnerPod(reqLogger logr.Logger, engineRunner *corev1.Pod) (*reconcileEngine, error) {
+func (r *ReconcileChaosEngine) checkEngineRunnerPod(ce *chaosTypes.EngineInfo, reqLogger logr.Logger) error {
+	if (len(ce.AppExperiments) == 0 || ce.AppUUID == "") && ce.Instance.Spec.AnnotationCheck == "true" {
+		return errors.New("application experiment list or UUID is empty")
+	}
+	var engineRunner *corev1.Pod
+	engineRunner, err := newGoRunnerPodForCR(ce)
```

Comment:
If we set `runner.type == ansible` then both  newGoRunnerPodForCR(ce) and  newAnsibleRunnerPodForCR(ce) will execute. Can you put this function call in else condition of this `if ce.Instance.Spec.Components.Runner.Type == "ansible"`

Reply:
Please add an issue for this, as its affecting BCH's and some assignment issues, btw we might fully deprecate `ansible runner` in this release.

Reply:
lets deprecate it/comment it out. There have been too many changes to the go runner which makes usage of the ansible runner in its current state untenable. The day we choose to merge runner with experiments, we can allow for users to create their own runners. As long as the present arch continues, we can use only go runner.